packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -115,6 +121,58 @@ class DatePickerInput extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    // Formats the selected date into a consistent value and sets input state.
+    if (props.value !== state.prevDate) {
+      const format = DateUtil.getDateFormat(props.intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' }));
+      const tempDate = DateUtil.convertToISO8601(props.value, format);
+      return {
+        day: tempDate.substring(8, 10),
+        month: tempDate.substring(5, 7),
+        year: tempDate.substring(0, 4),
+        prevDate: props.value,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {int} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateVariant - Date variant based on the currently supported date
+   *                      orders: 'MM-DD-YYYY', 'DD-MM-YYYY', or 'YYYY-MM-DD'.
+   */
+  moveFocusOnChange(inputValue, type, dateVariant) {
+    if (!this.state.prevDate) {
+      if (dateVariant === 'MM-DD-YYYY') {
```

Comment:
These checks seem bad. How do we handle all the local formats we provide? What about dutch? https://github.com/cerner/terra-framework/blob/master/packages/terra-date-picker/translations/nl.json#L2

Reply:
`nl` would fall under [this](https://github.com/cerner/terra-framework/blob/0e4f6a7773905b689fc5e94a2b0587f6e732ff3c/packages/terra-date-picker/src/DateUtil.jsx#L190). There are 3 actual date variations, but the input itself doesn't need to know whether it is `es` or `fr`, just what order they should be in.